### PR TITLE
 Fix SETUP-Inject-IP-Constants.ps1 and related testcases after root removal

### PR DESCRIPTION
**Pull request by asofron**
Affected Test Cases : 

1. NET-EXTERNAL
2. NET-INTERNAL
3. NET-CHANGE-INTERNAL-NIC
4. NET-JUMBO-FRAMES
5. NET-COPY-LARGE-FILE
6. NET-COPY-FILE-DIFFERENT-MTU
7. NET-COPY-BOTH-WAYS
8. NET-VXLAN
9. NET-MULTICAST
10. ETHTOOL-CHECK-STATISTICS

Verification : 
```
[LISAv2 Test Results Summary]
Test Run On           : 01/25/2019 14:57:40
VHD Under Test        :  ubuntu_18.04.1.vhdx
Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:0:11

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 NET-JUMBO-FRAMES                                                                  PASS                 7.53 




[LISAv2 Test Results Summary]
Test Run On           : 01/25/2019 15:11:42
VHD Under Test        : Centos_7.5_x64.vhdx
Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:0:13

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 NET-JUMBO-FRAMES                                                                  PASS                 8.39 




[LISAv2 Test Results Summary]
Test Run On           : 01/25/2019 15:10:59
VHD Under Test        : Centos_7.5_x64.vhdx
Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:0:14

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 NET-VXLAN                                                                         PASS                  9.6 


[LISAv2 Test Results Summary]
Test Run On           : 01/25/2019 15:31:52
VHD Under Test        :  ubuntu_18.04.1.vhdx
Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:0:10

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 NET-VXLAN                                                                         PASS                 7.97 





[LISAv2 Test Results Summary]
Test Run On           : 01/25/2019 15:09:45
VHD Under Test        :  ubuntu_18.04.1.vhdx
Total Test Cases      : 2 (2 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:0:16

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 NET-EXTERNAL                                                                      PASS                 5.01 
    2 NET-INTERNAL                                                                      PASS                 7.49 


[LISAv2 Test Results Summary]
Test Run On           : 01/25/2019 15:45:03
VHD Under Test        : Centos_7.5_x64.vhdx
Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:0:6

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 NET-INTERNAL                                                                      PASS                 3.82 


[LISAv2 Test Results Summary]
Test Run On           : 01/25/2019 15:52:21
VHD Under Test        : Centos_7.5_x64.vhdx
Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:0:5

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 NET-EXTERNAL                                                                      PASS                 3.25 




[LISAv2 Test Results Summary]
Test Run On           : 01/25/2019 15:47:49
VHD Under Test        :  ubuntu_18.04.1.vhdx
Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:0:5

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 NET-CHANGE-INTERNAL-NIC                                                           PASS                 3.12 




[LISAv2 Test Results Summary]
Test Run On           : 01/25/2019 15:48:27
VHD Under Test        : Centos_7.5_x64.vhdx
Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:0:5

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 NET-CHANGE-INTERNAL-NIC                                                           PASS                 3.21 



[LISAv2 Test Results Summary]
Test Run On           : 01/25/2019 16:12:28
VHD Under Test        : Centos_7.5_x64.vhdx
Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:0:9

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 NET-COPY-FILE-DIFFERENT-MTU                                                       PASS                 5.72 



[LISAv2 Test Results Summary]
Test Run On           : 01/25/2019 16:12:46
VHD Under Test        :  ubuntu_18.04.1.vhdx
Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:0:7

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 NET-COPY-FILE-DIFFERENT-MTU                                                       PASS                 4.92 





[LISAv2 Test Results Summary]
Test Run On           : 01/25/2019 16:24:20
VHD Under Test        :  ubuntu_18.04.1.vhdx
Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:0:8

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 NET-MULTICAST                                                                     PASS                 5.82 




[LISAv2 Test Results Summary]
Test Run On           : 01/25/2019 16:24:12
VHD Under Test        : Centos_7.5_x64.vhdx
Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:0:10

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 NET-MULTICAST                                                                     PASS                 6.92 





[LISAv2 Test Results Summary]
Test Run On           : 01/25/2019 16:42:38
VHD Under Test        : Centos_7.5_x64.vhdx
Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:0:9

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 NET-COPY-LARGE-FILE                                                               PASS                 5.38 



[LISAv2 Test Results Summary]
Test Run On           : 01/25/2019 16:42:22
VHD Under Test        :  ubuntu_18.04.1.vhdx
Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:0:7

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 NET-COPY-LARGE-FILE                                                               PASS                 4.88 

```
```
 [LISAv2 Test Results Summary]
 Test Run On           : 01/28/2019 15:59:48
 VHD Under Test        : ubuntu_18.04.1.vhdx
 Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
 Total Time (dd:hh:mm) : 0:0:13
 
    ID TestCaseName                                                                TestResult TestDuration(in minutes) 
 ---------------------------------------------------------------------------------------------------------------------
     1 ETHTOOL-CHECK-STATISTICS                                                          PASS                10.32 
```
```
 [LISAv2 Test Results Summary]
 Test Run On           : 01/28/2019 19:13:56
 VHD Under Test        : Centos_7.5_x64.vhdx
 Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
 Total Time (dd:hh:mm) : 0:0:14
 
    ID TestCaseName                                                                TestResult TestDuration(in minutes) 
 ---------------------------------------------------------------------------------------------------------------------
     1 ETHTOOL-CHECK-STATISTICS                                                          PASS                 8.26 ```

### CODE REVIEW

File: Testscripts/Windows/SETUP-Inject-IP-Constants.ps1 [hunk 1]
```diff
@@ -159,23 +159,23 @@ function Main {
...
+        Run-LinuxCmd -ip $Ipv4 -port $VMPort -username $VMUserName -password `
+            $VMPassword -command "chmod +x /home/$VMUserName/*.sh" -runAsSudo
+        Run-LinuxCmd -ip $vm2ipv4 -port $VMPort -username $VMUserName -password `
+            $VMPassword -command "chmod +x /home/$VMUserName/*.sh" -runAsSudo
+        $null = Run-LinuxCmd -ip $Ipv4 -port $VMPort -username $VMUserName -password `
+            $VMPassword -command "./enablePasswordLessRoot.sh /home/$VMUserName ; cp -rf /root/.ssh /home/$VMUserName" -runAsSudo
         # Copy keys from VM1 and setup VM2
         Copy-RemoteFiles -download -downloadFrom $Ipv4 -port $VMPort -files `
-            "/root/sshFix.tar" -username "root" -password $VMPassword -downloadTo $LogDir
+            " /home/$VMUserName/sshFix.tar" -username $VMUserName -password $VMPassword -downloadTo $LogDir
```
Comment (LiliDeng) [inline Testscripts/Windows/SETUP-Inject-IP-Constants.ps1]: blank before /home/$VMUserName/sshFix.tar

File: Testscripts/Windows/SETUP-Inject-IP-Constants.ps1 [hunk 2]
```diff
@@ -159,23 +159,23 @@ function Main {
...
         Copy-RemoteFiles -uploadTo $vm2ipv4 -port $VMPort -files "$LogDir\sshFix.tar" `
-            -username "root" -password $VMPassword -upload
-        $null = Run-LinuxCmd -ip $vm2ipv4 -port $VMPort -username "root" -password `
-            $VMPassword -command "./enablePasswordLessRoot.sh ; cp -rf /root/.ssh /home/$VMUserName"
+            -username $VMUserName -password $VMPassword -upload
+        $null = Run-LinuxCmd -ip $vm2ipv4 -port $VMPort -username $VMUserName -password `
+            $VMPassword -command "./enablePasswordLessRoot.sh /home/$VMUserName ; cp -rf /root/.ssh /home/$VMUserName" -runAsSudo
```
Comment (LiliDeng) [inline Testscripts/Windows/SETUP-Inject-IP-Constants.ps1]: ditto

File: Testscripts/Windows/SETUP-Inject-IP-Constants.ps1 (hunk 1 above)
Comment (asofron) [reply Testscripts/Windows/SETUP-Inject-IP-Constants.ps1]: FIxed

**Comment by lizzha** [comment]
@LiliDeng could you take a look and signoff if comments are resolved?

File: Testscripts/Linux/Ethtool-Check-Statistics.sh [hunk 3]
```diff
@@ -13,19 +13,22 @@
...
         LogErr "Unable to download netperf."
         return 1
     fi
-    tar -xvf netperf-2.7.0.tar.gz > /dev/null 2>&1
+    tar -xvf netperf-2.7.0.tar.gz  > /dev/null 2>&1
 
     #Get the root directory of the tarball
     downloadDir="/netperf-netperf-2.7.0"
-    rootDir=$(pwd)$downloadDir
+    homeDir="/home/${SUDO_USER}"
```
Comment (LiliDeng) [inline Testscripts/Linux/Ethtool-Check-Statistics.sh]: where does this value come from?

File: Testscripts/Linux/Ethtool-Check-Statistics.sh [hunk 4]
```diff
@@ -13,19 +13,22 @@
...
+    rootDir="${homeDir}/$downloadDir"
     LogMsg "rootDir = ${rootDir}"
+    LogMsg "Homedir = ${ Homedir}"
```
Comment (LiliDeng) [inline Testscripts/Linux/Ethtool-Check-Statistics.sh]: Homedir or homeDir?

File: Testscripts/Linux/Ethtool-Check-Statistics.sh (hunk 3 above)
Comment (asofron) [reply Testscripts/Linux/Ethtool-Check-Statistics.sh]: ${SUDO_USER} is populated on the env when sudo is run

File: Testscripts/Linux/Ethtool-Check-Statistics.sh (hunk 4 above)
Comment (asofron) [reply Testscripts/Linux/Ethtool-Check-Statistics.sh]: homeDir

File: Testscripts/Linux/Ethtool-Check-Statistics.sh [hunk 5]
```diff
@@ -13,19 +13,22 @@
...
+
 SendFile(){
     #Download netperf 2.7.0
     wget https://github.com/HewlettPackard/netperf/archive/netperf-2.7.0.tar.gz > /dev/null 2>&1
     if [ $? -ne 0 ]; then
         LogErr "Unable to download netperf."
         return 1
     fi
-    tar -xvf netperf-2.7.0.tar.gz > /dev/null 2>&1
+    tar -xvf netperf-2.7.0.tar.gz  > /dev/null 2>&1
```
Comment (chvalean) [inline Testscripts/Linux/Ethtool-Check-Statistics.sh]: remove extra space

File: Testscripts/Linux/Ethtool-Check-Statistics.sh [hunk 6]
```diff
@@ -13,19 +13,22 @@
...
 
     #Get the root directory of the tarball
     downloadDir="/netperf-netperf-2.7.0"
-    rootDir=$(pwd)$downloadDir
+    homeDir="/home/${SUDO_USER}"
+    rootDir="${homeDir}/$downloadDir"
     LogMsg "rootDir = ${rootDir}"
```
Comment (chvalean) [inline Testscripts/Linux/Ethtool-Check-Statistics.sh]: remove these 2 LogMsg entries, not relevant

File: Testscripts/Linux/Ethtool-Check-Statistics.sh [hunk 7]
```diff
@@ -13,19 +13,22 @@
...
+    LogMsg "homedir = ${homeDir}"
     cd ${rootDir}
 
     #Distro specific setup
```
Comment (chvalean) [inline Testscripts/Linux/Ethtool-Check-Statistics.sh]: space after #

File: Testscripts/Linux/Ethtool-Check-Statistics.sh [hunk 8]
```diff
@@ -156,18 +159,18 @@ SendFile(){
     fi
 
     LogMsg "Copy files to dependency vm: ${STATIC_IP2}"
-    scp -i "$HOME"/.ssh/"$SSH_PRIVATE_KEY" -v -o StrictHostKeyChecking=no ~/NET-Netperf-Server.sh ${remote_user}@[${STATIC_IP2}]: > /dev/null 2>&1
+    scp -i "$homeDir"/.ssh/"$SSH_PRIVATE_KEY" -v -o StrictHostKeyChecking=no ~/NET-Netperf-Server.sh ${remote_user}@[${STATIC_IP2}]: > /dev/null 2>&1
```
Comment (chvalean) [inline Testscripts/Linux/Ethtool-Check-Statistics.sh]: remove -v from all scp commands, no need for verbose

**Pull request merged**